test/test_complex.py

```diff
@@ -1,10 +1,24 @@
+import numpy as np
```

Comment:
You can't unconditionally import NumPy. What you can do is import TEST_NUMPY from common_utils.py (see how test_torch.py does this) and then import numpy if TEST_NUMPY is True

Reply:
(Although since you're only using NumPy for pi below, as best I can tell, you probably just want to switch to using math.pi.)